src/tracing.ts

```diff
@@ -20,10 +21,12 @@ export enum TracingMode {
 }
 
 export function enableTracing(service: Service, tracingMode: TracingMode, handlers: FunctionInfo[]) {
-  const provider = service.provider as any;
+  const provider = service.provider as Provider;
   if (tracingMode === TracingMode.XRAY || tracingMode === TracingMode.HYBRID) {
     provider.tracing = {
-      apiGateway: true,
+      apiGateway: provider.apiGateway?.restApiId
```

Comment:
Instead of leaving an empty key here, what would you think about changing the logic around?

We could do:
```
provider.tracing = { lambda: true }

if (!provider.apiGateway?.restApiId) {
  provider.tracing.apiGateway = true
}
```

Thoughts?

Reply:
Yes, that would be ideal, but unfortunately, the problem is with the type not allowing to do that.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b6e1b73743d2fdf87d61b0e667e015835aa021f6/types/serverless/plugins/aws/provider/awsProvider.d.ts#L269

I thought it will be a lesser evil to typecast `undefined` to `any` rather than to typecast the whole object or resort to `@ts-ignore`.

I've submitted PR to the types repo https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60489

Reply:
Ha! Okay, thanks for that context. I doubt that the Serverless Framework maintainers will change this behavior to interpret `undefined` as something else, so I think we can accept this patch. Thanks again!